cmd/npmExecuteLint.go

```diff
@@ -0,0 +1,170 @@
+	response, err := u.client.SendRequest(http.MethodGet, configURL, nil, nil, nil)
+	if err != nil {
+		log.Entry().Warnf("failed to download general purpose configuration: %v", err)
+		return
+	}
+
+	defer response.Body.Close()
+
+	content, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		log.Entry().Warnf("error while reading the general purpose configuration: %v", err)
+		return
+	}
+
+	err = u.FileWrite(filepath.Join(".pipeline", ".eslintrc.json"), content, os.ModePerm)
+	if err != nil {
+		log.Entry().Warnf("failed to write .eslintrc.json file to .pipeline/: %w", err)
+	}
+}
+
+func npmExecuteLint(config npmExecuteLintOptions, telemetryData *telemetry.CustomData) {
+	utils := newLintUtilsBundle()
+	npmExecutorOptions := npm.ExecutorOptions{DefaultNpmRegistry: config.DefaultNpmRegistry, SapNpmRegistry: config.SapNpmRegistry, ExecRunner: utils.getExecRunner()}
+	npmExecutor := npm.NewExecutor(npmExecutorOptions)
+
+	err := runNpmExecuteLint(npmExecutor, utils, &config)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runNpmExecuteLint(npmExecutor npm.Executor, utils lintUtils, config *npmExecuteLintOptions) error {
+	packageJSONFiles := npmExecutor.FindPackageJSONFiles()
+	packagesWithCiLint, _ := npmExecutor.FindPackageJSONFilesWithScript(packageJSONFiles, "ci-lint")
+
+	if len(packagesWithCiLint) > 0 {
+		err := runCiLint(npmExecutor, config.FailOnError)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := runDefaultLint(npmExecutor, utils, config.FailOnError)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func runCiLint(npmExecutor npm.Executor, failOnError bool) error {
+	runScripts := []string{"ci-lint"}
+	runOptions := []string{"--silent"}
+
+	err := npmExecutor.RunScriptsInAllPackages(runScripts, runOptions, false)
+	if err != nil {
+		if failOnError {
+			return fmt.Errorf("ci-lint script execution failed with error: %w. This might be the result of severe linting findings, or some other issue while executing the script. Please examine the linting results in the UI, the cilint.xml file, if available, or the log above. ", err)
+		}
+	}
+	return nil
+}
+
+func runDefaultLint(npmExecutor npm.Executor, utils lintUtils, failOnError bool) error {
+	execRunner := utils.getExecRunner()
+	eslintConfigs := findEslintConfigs(utils)
+
+	err := npmExecutor.SetNpmRegistries()
+	if err != nil {
+		log.Entry().Warnf("failed to set npm registries before running default lint: %v", err)
+	}
+
+	// If the user has ESLint configs in the project we use them to lint existing JS files. In this case we do not lint other types of files,
+	// i.e., .jsx, .ts, .tsx, since we can not be sure that the provided config enables parsing of these file types.
+	if len(eslintConfigs) > 0 {
+		for i, config := range eslintConfigs {
+			dir := filepath.Dir(config)
+			if dir == "." {
+				// Ignore possible errors when invoking ci-lint script to not fail the pipeline based on linting results
```

Comment:
```suggestion
				// Ignore possible errors when invoking lint script to not fail the pipeline based on linting results
```

Reply:
Yes you are right thats a copy paste error and actually the comment is not valid anymore, since there is the `failOnError` flag now.